### ISSUE TITLE
[List of working instances] Invalid Request: This helps protect our community. Learn more

### ISSUE THREAD

**Issue by ghost** [Siriusmart/youtube-tui#60]
Hey i got this Error, like on #57 its a invidous-youtube Error. i Found a list for invidous instances (`https://docs.invidious.io/instances/`) and i tried every of them so you dont have to!

### Working
[invidious.darkness.services](https://invidious.darkness.services/) 🇺🇸  ✅ Works perfectly
[invidious.fdn.fr](https://invidious.fdn.fr/) 🇫🇷 ✅ Works perfectly
[invidious.flokinet.to](https://invidious.flokinet.to/) 🇷🇴  ✅ Works perfectly

### Not Working
[yewtu.be](https://yewtu.be/) 🇩🇪 ❌ youtube-tui Parsing Error
[yt.artemislena.eu](https://yt.artemislena.eu/)🇩🇪 ❌ youtube-tui Parsing Error
[invidious.privacydev.net](https://invidious.privacydev.net/) 🇫🇷 ❌ youtube-tui Parsing Error
[invidious.privacydev.net](https://invidious.privacydev.net/) 🇫🇷 ❌ youtube-tui Parsing Error
[iv.melmac.space](https://iv.melmac.space/) 🇩🇪 ❌ youtube-tui Parsing Error
[inv.nadeko.net](https://inv.nadeko.net/) 🇨🇱 ❌ youtube-tui Parsing Error
[inv.tux.pizza](https://inv.tux.pizza/) 🇺🇸 ❌ youtube-tui Parsing Error
[invidious.protokolla.fi](https://invidious.protokolla.fi/) 🇫🇮  ❌ youtube-tui Parsing Error
[invidious.private.coffee](https://invidious.private.coffee/) 🇦🇹  ❌ youtube-tui Parsing Error
[yt.drgnz.club](https://yt.drgnz.club/) 🇨🇿  ❌ youtube-tui Parsing Error
[iv.datura.network](https://iv.datura.network/) 🇫🇮 ❌ youtube-tui Parsing Error
[invidious.perennialte.ch](https://invidious.perennialte.ch/) 🇦🇺  ❌ youtube-tui Parsing Error
[invidious.drgns.space](https://invidious.drgns.space/) 🇺🇸 ❌ youtube-tui Parsing Error
[invidious.nerdvpn.de](https://invidious.nerdvpn.de/) 🇺🇦 ❌ youtube-tui Parsing Error
[invidious.jing.rocks](https://invidious.jing.rocks/) 🇯🇵 ❌ youtube-tui Parsing Error
[invidious.privacyredirect.com](https://invidious.privacyredirect.com/) 🇫🇮  ❌ youtube-tui Parsing Error
[invidious.reallyaweso.me](https://invidious.reallyaweso.me/) 🇩🇪 ❌ youtube-tui Parsing Error
[invidious.materialio.us](https://invidious.materialio.us/) 🇳🇿 ❌ youtube-tui Parsing Error
[inv.in.projectsegfau.lt](https://inv.in.projectsegfau.lt/) 🇮🇳 ❌ youtube-tui Parsing Error
[invidious.incogniweb.net](https://invidious.incogniweb.net/) 🇺🇸 ❌ youtube-tui Parsing Error

`(you could make a function to check those domains automatically)`

#### Disclaimer
```
i'm using a VPN, maybe my IP is blocked but those are the invidous instances that work for me.
Maybe some work for you and some may not!
```

:heart:

**Comment by Siriusmart** [comment]
tysm for this, i can prob write an uptimerobot-like page for this kind of stuff

put for now, i'll just pin this page

**Comment by Siriusmart** [comment]
im doing an invidious health check site, that runs differently from api.invidious.io in a few different ways
- health is given by 4 factors: ability to load videos, playlists, channels, and searches, if the instance can do all four, then it may be flagged as a "hot" instance, meaning users will be recommended that instance
- users can add instances by clicking a button, which is part of the frontend i will do in a bit
- for each region, americas, europe and asia, there may be 10 or a 100 sites stashed (stored and not used), only 4 instances per region can be "hot" at the same time, as im aware that google is keeping a close eye on any invidious listings to block them. I don't want to make it too easy for them to blacklist all instances

so basically, instead of handing out address of all instances, which may make the instances easily blocked, the site only gives 1 instances to the user and hides all other instances, currently working on the UI

![image](https://github.com/user-attachments/assets/c1778c72-cde1-4be7-b3dc-9b541c5f7c11)

**Comment by Siriusmart** [comment]
[i found more instances](https://vidiup.siri.sh/)

**Comment by v0id86** [comment]
I think i might be doing something wrong. If i change the invidous instance on main.yml, instead of getting the "Invalid request: This helps protect our community." i just get "Invalid type: null, expected a string". Any ideas?

**Comment by Siriusmart** [comment]
u didnt do anything wrong, im working on to fixing it, but i havent touched this project in 2 years, so im working on bringing everything up to date at the moment

**Comment by v0id86** [comment]
> u didnt do anything wrong, im working on to fixing it, but i havent touched this project in 2 years, so im working on bringing everything up to date at the moment

Oh ok. Well i hope you're able to do it, i really enjoy it. I know close to nothing about programming, but if there's any other way i can help just let me know.

**Comment by Siriusmart** [comment]
> Hey i got this Error, like on #57 its a invidous-youtube Error. i Found a list for invidous instances (`https://docs.invidious.io/instances/`) and i tried every of them so you dont have to! ...

ok i realised what had happened, so i was updating the `invidious` crate to try and fix the issues, and all the tests on the `invidious` crate are passing, but... i erm forgot to make youtube-tui use the up to date version of `invidous` nothing is fixed.

anyhow, there should be no more parsing errors for now, if the instance is not blocked, then it works on youtube-tui

**Comment by Siriusmart** [comment]
> Oh ok. Well i hope you're able to do it, i really enjoy it. I know close to nothing about programming, but if there's any other way i can help just let me know.

try installing with 
```sh
cargo install --git https://github.com/Siriusmart/youtube-tui
```

also, i dont know if the updated mpv bindings is gonna work, because as i am aware some distros have slightly out of date versions of mpv, and thats the reason why i kept the old mpv bindings for the crates.io version as i dont know if the new one will work for everyone (whereas the old one doesnt work for me)

if u would tell me ur distro and mpv version, that would be great
```sh
mpv --version
```

**Comment by v0id86** [comment]
> > Oh ok. Well i hope you're able to do it, i really enjoy it. I know close to nothing about programming, but if there's any other way i can help just let me know.
> 
> try installing with
> 
> ```shell
> cargo install --git https://github.com/Siriusmart/youtube-tui
> ```
> 
> also, i dont know if the updated mpv bindings is gonna work, because as i am aware some distros have slightly out of date versions of mpv, and thats the reason why i kept the old mpv bindings for the crates.io version as i dont know if the new one will work for everyone (whereas the old one doesnt work for me)
> 
> if u would tell me ur distro and mpv version, that would be great
> 
> ```shell
> mpv --version
> ```

So, i installed it using the cargo install you provided and after changing the invidious instance everything seems to be working fine. 

Im running Arch wiht KDE Plasma (Wayland) and $ mpv --version
mpv v0.38.0-dirty Copyright © 2000-2024 mpv/MPlayer/mplayer2 projects
 built on Sep  4 2024 19:59:01
libplacebo version: v7.349.0
FFmpeg version: n7.0.2
FFmpeg library versions:
   libavutil       59.8.100
   libavcodec      61.3.100
   libavformat     61.1.100
   libswscale      8.1.100
   libavfilter     10.1.100
   libswresample   5.1.100

**Comment by ruvido** [comment]
installed from alpine repo and get the "invalid request", cannot understand what's wrong exactly?

**Comment by Siriusmart** [comment]
> installed from alpine repo and get the "invalid request", cannot understand what's wrong exactly?

i dont really know how the alpine packages are maintained, but i've just push an update to crates.io, so if the alpine packages are updated automatically, it should be updated shortly

previously only the git version is working, but now that its on crates.io, you can install it with

```sh
cargo install youtube-tui
```

you might want to change `~/.config/youtube-tui/main.yml` to use a different instance, i dont think vid.puffyan.us is working

**Comment by Siriusmart** [comment]
seeing this to be mostly resolved, i'll close the issue

**Comment by kqvanity** [comment]
@Siriusmart 
I'm facing the same error message, but only at certain videos.
The issue is only present at my [local production invidious instance](https://docs.invidious.io/installation/#docker-compose-method-production). Basically connecting it to `localhost:3000` port locally

**Comment by Siriusmart** [comment]
> [@Siriusmart](https://github.com/Siriusmart) I'm facing the same [error](https://github.com/Siriusmart/youtube-tui/issues/60#issuecomment-2347122723) message, but only at certain videos. The issue is only present at my [local production invidious instance](https://docs.invidious.io/installation/#docker-compose-method-production). Basically connecting it to `localhost:3000` port locally

If this is only present on ur local instance, can u send me the JSON file at `localhost:3000/api/v1/videos/THE_VIDEO_ID`?
